Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.h

```diff
@@ -0,0 +1,139 @@
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+/*=========================================================================
+ *
+ *  Portions of this file are subject to the VTK Toolkit Version 3 copyright.
+ *
+ *  Copyright (c) Ken Martin, Will Schroeder, Bill Lorensen
+ *
+ *  For complete copyright, license and disclaimer of warranty information
+ *  please refer to the NOTICE file at the top of the ITK source tree.
+ *
+ *=========================================================================*/
+
+#ifndef itkTriangleMeshCurvatureCalculator_h
+#define itkTriangleMeshCurvatureCalculator_h
+
+#include "itkMesh.h"
+
+namespace itk
+{
+template <typename TInputMesh>
+class ITK_TEMPLATE_EXPORT TriangleMeshCurvatureCalculator : public Object
+{
+public:
+  ITK_DISALLOW_COPY_AND_MOVE(TriangleMeshCurvatureCalculator);
+
+  /**
+   * Standard class type aliases.
+   */
+  using Self = TriangleMeshCurvatureCalculator;
+
+  /**
+   * Standard "Superclass" type alias.
+   */
+  using Superclass = Object;
+
+  /**
+   * Smart pointer type alias support
+   */
+  using Pointer = SmartPointer<Self>;
+  using ConstPointer = SmartPointer<const Self>;
+
+  /**
+   * Method for creation through the object factory.
+   */
+  itkNewMacro(Self);
+
+  /** Run-time type information (and related methods). */
+  itkTypeMacro(TriangleMeshCurvatureCalculator, Object);
+
+  /**
+   * Convenient type alias for this Calculator.
+   */
+  using InputMeshType = TInputMesh;
+  using MeshConstPointer = typename InputMeshType::ConstPointer;
+  using MeshPointType = typename InputMeshType::PointType;
+  using MeshPointIdConstIterator = typename InputMeshType::CellTraits::PointIdConstIterator;
+  using MeshPointsContainerConstPointer = typename InputMeshType::PointsContainerConstPointer;
+  using MeshPointDataContainer = typename InputMeshType::PointDataContainer;
+  using MeshPointDataContainerPointer = typename InputMeshType::PointDataContainerPointer;
+  using CellsContainerConstPointer = typename InputMeshType::CellsContainerConstPointer;
+  using CellsContainerConstIterator = typename InputMeshType::CellsContainer::ConstIterator;
+  using CellType = typename InputMeshType::CellType;
+
+  /**
+   * Different modes of operation. Use these to specify
+   * how to extract the regions. Currently only Guassian curvature is supported.
+   * Using enum for future extension to Mean, Max and Min curvature.
+   */
+  enum
```

Comment:
ITK is using strongly typed enums. Please change this to such type, add the corresponding ostream overload, and exercise it in the test.